sljit_src/sljitProtExecAllocator.c

```diff
@@ -96,51 +102,47 @@ struct chunk_header {
 #endif
 #endif
 
-#if !(defined(__NetBSD__) && defined(MAP_REMAPDUP))
 int mkostemp(char *template, int flags);
 char *secure_getenv(const char *name);
 
 static SLJIT_INLINE int create_tempfile(void)
 {
 	int fd;
-
+	struct stat st;
 	char tmp_name[256];
-	size_t tmp_name_len;
+	size_t tmp_name_len = 0;
 	char *dir;
 	size_t len;
 
-#ifdef P_tmpdir
-	len = (P_tmpdir != NULL) ? strlen(P_tmpdir) : 0;
```

Comment:
P_tmpdir is a macro (AKA a string literal) not a (char *) therefore couldn't be NULL